.github/workflows/benchmark.yml

```diff
@@ -52,8 +52,8 @@ jobs:
           # Install typing-extensions separately whilst the `--extra-index-url` in `requirements-ci.txt`
           # doesn't have an up-to-date typing-extensions, see
           # https://github.com/astral-sh/uv/issues/6028#issuecomment-2287232150
-          uv pip install -U typing-extensions 
-          uv pip install --compile-bytecode -r requirements-dev.txt -r requirements-ci.txt --verbose
+          uv pip install -U typing-extensions
+          uv pip install --compile-bytecode -r requirements-dev.txt -r requirements-ci.txt --verbose --index-strategy=unsafe-best-match
```

Comment:
The `--extra-index-url https://download.pytorch.org/whl/cpu` entry in `requirements-ci.txt` caueses issues for NumPy installation under Python 3.13. Specifying this index strategy makes it so that NumPy is downloaded from the regular PyPI registry rather than the PyTorch registry.